ui/panels-plugin/src/plugins/bar-chart/BarChartPanel.tsx

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { BarChart, BarChartData } from '@perses-dev/components';
+import { Box, Stack, Skeleton } from '@mui/material';
+import { useMemo } from 'react';
+import { CalculationsMap } from '@perses-dev/core';
+import { useDataQueries, PanelProps } from '@perses-dev/plugin-system';
+import { BarChartOptions } from './bar-chart-model';
+import { calculatePercentages, sortSeriesData } from './utils';
+
+const PADDING = 24;
+
+export type BarChartPanelProps = PanelProps<BarChartOptions>;
+
+export function BarChartPanel(props: BarChartPanelProps) {
+  const {
+    spec: { calculation, unit, sort, mode },
+    contentDimensions,
+  } = props;
+
+  const { queryResults, isLoading, isFetching } = useDataQueries(); // gets data queries from a context provider, see DataQueriesProvider
+
+  const barData: BarChartData[] = useMemo(() => {
+    if (queryResults[0]?.data === undefined) {
+      return [];
+    }
+    const seriesData: BarChartData[] = [];
+    for (const timeSeries of queryResults[0].data.series) {
```

Comment:
Actually i think we need to loop through `queryResults` here since you can have multiple queries, and each query can also have multiple series

Reply:
You can find an example using the LegendExample panel in Benchmark dashboard

Reply:
Updated! f9dc6a03c675261b3259dba007af55aabe3ec478